roles/openshift_storage_glusterfs/tasks/update_topology.yml

```diff
@@ -0,0 +1,31 @@
+---
+# This taskfile is called when adding new nodes doing node and master
+# scaleup play.
+- import_tasks: mktemp.yml
+
+# l_gluster_reload_topo passed in via add_hosts.yml
+- when: l_gluster_reload_topo | default(True)
+  block:
+  - import_tasks: glusterfs_config_facts.yml
+  - import_tasks: label_nodes.yml
+  - import_tasks: heketi_pod_check.yml
+  - import_tasks: get_heketi_key.yml
+  - import_tasks: wait_for_pods.yml
+  - import_tasks: heketi_load.yml
+    when:
+    - glusterfs_nodes | default([]) | count > 0
+
+# l_gluster_registry_reload_topo passed in via add_hosts.yml
+- when: l_gluster_registry_reload_topo | default(True)
+  block:
+  - import_tasks: glusterfs_registry_facts.yml
+  - import_tasks: label_nodes.yml
+  - import_tasks: heketi_pod_check.yml
+  - import_tasks: get_heketi_key.yml
+  - import_tasks: wait_for_pods.yml
+  - import_tasks: heketi_load.yml
+    when:
+    - glusterfs_nodes | default([]) | count > 0
+    - "'glusterfs' not in groups or glusterfs_nodes != groups.glusterfs"
```

Comment:
These conditionals should be against `glusterfs_registry_nodes`.

Reply:
@jarrpa I'm not sure that's accurate.  the glusterfs_nodes is generated by the set_fact task files.  They're not actual groups, and there is no mention of glusterfs_registry_nodes.

Reply:
...oh, right. My head was in the wrong file.

In that case, you shouldn't be doing conditional checks on `glusterfs_nodes` at all. Since you're not calling `glusterfs_config_facts.yml` or `glusterfs_registry_facts.yml` the `glusterfs_nodes` variable wouldn't be set. Both above and below should only be by group names.